ipfshttpclient/exceptions.py

```diff
@@ -97,27 +102,44 @@ class DecodingError(EncoderError):
 	"""Raised when decoding a byte string to a Python object has failed due to
 	some problem with the input data."""
 	__slots__ = ("original",)
-	#original: Exception
+
+	original: Exception
 	
 	def __init__(self, encoder_name: str, original: Exception) -> None:
-		self.original = original  # type: Exception
+		self.original = original
 		
 		super().__init__("Object decoding error: {}".format(original), encoder_name)
 
 
+##################
+# filescanner.py #
+##################
+
+class MatcherSpecInvalidError(TypeError):
+	"""
+	An attempt was made to build a matcher using matcher_from_spec, but an invalid
+	specification was provided.
+	"""
+
+	def __init__(self, matcher_class: type, invalid_spec: ty.Any) -> None:
+		super().__init__(
+			f"Don't know how to create a {matcher_class.__name__} from spec {invalid_spec!r}"
+		)
```

Comment:
This seems a bit nonsensical to me. Won't always `matcher_class.__name__ == "Matcher"`?